### PR TITLE
Making the spawn menu functional, making goal markers include goal name, fixing Harmony warnings

### LINKED ISSUE #2: Make a submenu for choosing points A and B
when you choose "Nomai Grand Prix" you should go to a submenu that looks like "The Interloper -> Brittle Hollow Gravity Cannon". The locations should be randomly prepopulated, and you should be able to use the direction buttons to change them (up and down change them in a list, left and right select point A or B). Pressing start or A starts the mod.

### LINKED ISSUE #71: Randomize Spawn & Goal button should do what it says
Currently (due to #70), this button just resets the existing speedrun instead of randomizing it.

### PR DESCRIPTION
**Pull request by cmaines**
These changes do the following:
- Removes the "Pingus Mode" button and makes the "Nomai Grand Prix" button open the spawn menu
- Makes submitting the spawn menu start a speedrun with the selected spawns
- Adds the name of the goal to the markers on the HUD and map
- Eliminates Harmony patch warnings
- Renames the "RESET SPAWN AND GOAL" button to "RANDOMIZE SPAWN AND GOAL"
- Makes the Randomize Spawn And Goal button do what it says

Testing:
- Verified that pingus mode button is gone (and quit button works again)
- Verified that nomai grand prix button starts the spawn menu
- Verified that submitting the spawn menu starts a speedrun with the selected spawns
- Verified that the HUD and map markers include the name of the goal
- Verified that you can quit to the main menu and start a new run
- Verified that clicking Randomize Spawn And Goal does what it says

### CODE REVIEW

File: OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs [hunk 1]
```diff
@@ -170,14 +136,15 @@ private void CreateTimer()
 
         private void HandleBasicWarp(PlayerSpawner spawner, SpawnPoint[] spawnPoints)
         {
-            if (_spawnPointName == null || _goalPointName == null)
+            if (SpeedrunState.INSTANCE.SpawnPointId == null || SpeedrunState.INSTANCE.GoalPointId == null)
```
Comment (cmaines) [inline OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs:139]: Oops this should be deleted, but it's effectively dead code at the moment. Will fix in a followup.

**Comment by cmaines** [comment]
Created #71 to fix the regression this introduces & assigned to myself.

**Comment by bucketon** [comment]
LGTM pending two changes we talked about on discord (use singleton shortcut on getters, cache whole spawn point struct instead of individual fields)

**Comment by bucketon** [comment]
LGTM!

**Pull request merged**